presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -298,12 +299,16 @@ public PlanOptimizers(Metadata metadata, SqlParser sqlParser, FeaturesConfig fea
             builder.add(new AddExchanges(metadata, sqlParser));
         }
 
-        builder.add(new PickLayout(metadata));
+        builder.add(
+                new IterativeOptimizer(
+                        stats,
+                        ImmutableList.of(new com.facebook.presto.sql.planner.optimizations.PickLayout(metadata)),
+                        ImmutableSet.of(new PickTableLayoutForPredicate(metadata), new PickTableLayoutWithoutPredicate(metadata))));
```

Comment:
I think we need some kind of test which will test these two rules together to make sure they work together and do not cycle. It could be done in TestLogicalPlanner`.